app/controllers/api/v1/messages_controller.rb

```diff
@@ -21,15 +21,15 @@ def index
   end
 
   def create
-    @device_message = DeviceMessage.find_by(device_id: params[:device_message][:device_id],
-                                            message_id: params[:device_message][:message_id])
-    @device_message = DeviceMessage.create!(device_message_params) if @device_message.nil?
+    device = Device.find_by_token(params[:device_message][:token])
+    @device_message = device.device_messages.find_by(message_id: params[:device_message][:message_id])
```

Comment:
@nrailsy97 this will only create one device_message per message. So if SAF sends out one message to 100 users we will only be recording the first person to read the message, we have to search by token also here.